From 841cc9fab750553eadb2b1ded39962b4ab3d6cba Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Tue, 7 Oct 2025 20:37:27 +0300
Subject: [PATCH 346/553] tst_QObject: port away from QPair

Drive-by, use initializer list instead of QList::operator<<().

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: I4ce6a890cf76726638eb96fe932c8a0331ac2ba6
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 789da74878688c93f7e208db9b91e32274aa93f3)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../corelib/kernel/qobject/tst_qobject.cpp    | 64 +++++++------------
 1 file changed, 22 insertions(+), 42 deletions(-)

diff --git a/tests/auto/corelib/kernel/qobject/tst_qobject.cpp b/tests/auto/corelib/kernel/qobject/tst_qobject.cpp
index 70c3d9843d2..1be228054b8 100644
--- a/tests/auto/corelib/kernel/qobject/tst_qobject.cpp
+++ b/tests/auto/corelib/kernel/qobject/tst_qobject.cpp
@@ -3266,7 +3266,7 @@ class EventSpy : public QObject
     Q_OBJECT
 
 public:
-    typedef QList<QPair<QObject *, QEvent::Type> > EventList;
+    using EventList = QList<std::pair<QObject *, QEvent::Type>>;
 
     EventSpy(QObject *parent = nullptr)
         : QObject(parent)
@@ -3285,7 +3285,7 @@ public:
 
     bool eventFilter(QObject *object, QEvent *event) override
     {
-        events.append(qMakePair(object, event->type()));
+        events.append({object, event->type()});
         thisCounter = ++s_eventSpyCounter;
         return false;
     }
@@ -3309,9 +3309,7 @@ void tst_QObject::childEvents()
 
         QCoreApplication::processEvents();
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&object, QEvent::Type(QEvent::User + 1));
+        expected = {{&object, QEvent::Type(QEvent::User + 1)}};
         QCOMPARE(spy.eventList(), expected);
     }
 
@@ -3329,19 +3327,15 @@ void tst_QObject::childEvents()
 
         QCoreApplication::postEvent(&object, new QEvent(QEvent::Type(QEvent::User + 2)));
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&object, QEvent::ChildAdded)
-            << qMakePair(&object, QEvent::ChildAdded);
+        expected = { {&object, QEvent::ChildAdded},
+                     {&object, QEvent::ChildAdded} };
         QCOMPARE(spy.eventList(), expected);
         spy.clear();
 
         QCoreApplication::processEvents();
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&object, QEvent::Type(QEvent::User + 1))
-            << qMakePair(&object, QEvent::Type(QEvent::User + 2));
+        expected = { {&object, QEvent::Type(QEvent::User + 1)},
+                     {&object, QEvent::Type(QEvent::User + 2)} };
         QCOMPARE(spy.eventList(), expected);
     }
 
@@ -3361,20 +3355,16 @@ void tst_QObject::childEvents()
 
         QCoreApplication::postEvent(&object, new QEvent(QEvent::Type(QEvent::User + 2)));
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&object, QEvent::ChildAdded)
-            << qMakePair(&object, QEvent::ChildAdded)
-            << qMakePair(&object, QEvent::ChildRemoved);
+        expected = { {&object, QEvent::ChildAdded},
+                     {&object, QEvent::ChildAdded},
+                     {&object, QEvent::ChildRemoved} };
         QCOMPARE(spy.eventList(), expected);
         spy.clear();
 
         QCoreApplication::processEvents();
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&object, QEvent::Type(QEvent::User + 1))
-            << qMakePair(&object, QEvent::Type(QEvent::User + 2));
+        expected ={ {&object, QEvent::Type(QEvent::User + 1)},
+                    {&object, QEvent::Type(QEvent::User + 2)} };
         QCOMPARE(spy.eventList(), expected);
     }
 }
@@ -3405,10 +3395,8 @@ void tst_QObject::parentEvents()
 
         QCoreApplication::processEvents();
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&child, QEvent::Type(QEvent::User + 1))
-            << qMakePair(&child, QEvent::Type(QEvent::User + 2));
+        expected = { {&child, QEvent::Type(QEvent::User + 1)},
+                     {&child, QEvent::Type(QEvent::User + 2)} };
         QCOMPARE(spy.eventList(), expected);
     }
 
@@ -3429,21 +3417,17 @@ void tst_QObject::parentEvents()
 
         QCoreApplication::postEvent(&child, new QEvent(QEvent::Type(QEvent::User + 2)));
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&child, QEvent::ParentAboutToChange)
-            << qMakePair(&child, QEvent::ParentChange)
-            << qMakePair(&child, QEvent::ParentAboutToChange)
-            << qMakePair(&child, QEvent::ParentChange);
+        expected = { {&child, QEvent::ParentAboutToChange},
+                     {&child, QEvent::ParentChange},
+                     {&child, QEvent::ParentAboutToChange},
+                     {&child, QEvent::ParentChange} };
         QCOMPARE(spy.eventList(), expected);
         spy.clear();
 
         QCoreApplication::processEvents();
 
-        expected =
-            EventSpy::EventList()
-            << qMakePair(&child, QEvent::Type(QEvent::User + 1))
-            << qMakePair(&child, QEvent::Type(QEvent::User + 2));
+        expected = { {&child, QEvent::Type(QEvent::User + 1)},
+                     {&child, QEvent::Type(QEvent::User + 2)} };
         QCOMPARE(spy.eventList(), expected);
     }
 #else
@@ -3462,9 +3446,7 @@ void tst_QObject::installEventFilter()
 
     // nothing special, should just work
     QCoreApplication::sendEvent(&object, &event);
-    expected =
-        EventSpy::EventList()
-        << qMakePair(&object, QEvent::User);
+    expected = { {&object, QEvent::User} };
     QCOMPARE(spy.eventList(), expected);
     spy.clear();
 
@@ -3477,9 +3459,7 @@ void tst_QObject::installEventFilter()
     // move it back, and the filter works again
     spy.moveToThread(object.thread());
     QCoreApplication::sendEvent(&object, &event);
-    expected =
-        EventSpy::EventList()
-        << qMakePair(&object, QEvent::User);
+    expected = { {&object, QEvent::User} };
     QCOMPARE(spy.eventList(), expected);
     spy.clear();
 
-- 
2.51.2

